### PR TITLE
[test] additional tests for devbox init and run

### PR DESCRIPTION
**Pull request by mohsenari**
## Summary
Title says it.

## How was it tested?
Ran the tests locally and it succeeded

### CODE REVIEW

File: examples/testdata/testframework/impl/framework.go [hunk 1]
```diff
@@ -18,27 +18,54 @@ type TestDevbox struct {
...
+func (td *TestDevbox) Info(pkg string, markdown bool) (string, error) {
+	cmd := boxcli.InfoCmd()
+	output, err := runCmd(cmd, []string{pkg})
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+	return string(output), nil
+}
+
+func (td *TestDevbox) Init() (string, error) {
```
Comment (mohsenari) [inline examples/testdata/testframework/impl/framework.go:66]: this is the only function I added. The rest of the changes in this file are due to re-organizing the order of funcs to the same order that they are listed in interface.go

File: examples/testdata/commands/init/init_test.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	err = exec.Command("touch", "requirements.txt").Run()
+	assert.NoError(t, err)
+	output, err := td.Init()
+	assert.NoError(t, err)
+	assert.FileExists(t, "devbox.json")
+	assert.Contains(t, output, "We detected extra packages you may need.")
+	assert.Contains(t, output, "devbox add")
+	assert.Contains(t, output, "nodejs")
+	assert.Contains(t, output, "python3")
+}
```
Comment (savil) [inline examples/testdata/commands/init/init_test.go:33]: nice!

File: examples/testdata/commands/run/run_test.go [hunk 3]
```diff
@@ -16,3 +16,17 @@ func TestRun(t *testing.T) {
 	_, err = td.Run("test1")
 	assert.NoError(t, err)
 }
+
+func TestRunCommand(t *testing.T) {
+	td := testframework.Open()
+	err := td.SetDevboxJson("devbox.json")
+	assert.NoError(t, err)
+
+	_, err = td.Run("ls > test.txt")
```
Comment (savil) [inline examples/testdata/commands/run/run_test.go:25]: is this doing whatever the current functionality is to do what `devbox shell -- <command>` used to do?

**Review by savil — APPROVED** [review]

File: examples/testdata/commands/run/run_test.go (hunk 3 above)
Comment (mohsenari) [reply examples/testdata/commands/run/run_test.go:25]: @savil basically yes. @ipince can give more context on the difference

**Pull request merged**